Fix config files for demultiplexing

beatrizsavinhas opened the pull request:
### This PR adds/fixes:

- Fix config file for hiseqx demultiplexing

**How to prepare for test**:
- [x] ssh to hasta
- [x] install on stage:
`bash servers/resources/hasta.scilifelab.se/update-demux-stage.sh fixing-demux-config-file`

**How to test**:
- [x] login to hasta
- [x] activate stage
- [x] copy flow cell to stage
- [x] test demultiplexing with master
- [x] installing branch in stage
- [x] test demultiplexing

**Expected test outcome**:
- [X] Check that SampleSheet.csv is generated properly
- [X] Check if demux is started correctly
- [X] Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by Beatriz
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

Commit:
Fix config file for hiseqx demultiplexing

Commit:
Fixed demux 2500 and novaseq

karlnyr commented on scripts/hiseqx/xdemuxtiles.batch:
```diff
@@ -10,6 +10,8 @@
 
 set -e -o pipefail
 
+useprod
+
```
I dont think we need this since we do the check in line 27-30

beatrizsavinhas replied:
You're right. Forgot to delete this!

karlnyr commented on scripts/novaseq/demux-novaseq.bash:
```diff
@@ -15,14 +15,15 @@ PROJECTLOG=${4?'projectlog needed'}
     SLURM_ACCOUNT=production
 fi
 
+SCRIPT_DIR=/home/proj/${ENVIRONMENT}/bin/git/demultiplexing/scripts/novaseq/
+
```
Dont think this was needed - we also don't actually use this script since we do the demultiplexing of novaseq through `cg`

beatrizsavinhas replied:
Removed this from the pull request.

karlnyr reviewed (commented):
Nice! We should construct some tests for this - perhaps by trying to copy the problematic flowcell to the stage position and then use the automation scripts to try to start the demultiplexing.

Commit:
Answering Kalle's comments.

beatrizsavinhas commented on scripts/hiseqx/xdemuxtiles.batch:
```diff
@@ -21,19 +23,13 @@ ENVIRONMENT=${4?'Environment'}
-
+usestage
 if [[ ${ENVIRONMENT} == 'production' ]]; then
     SLURM_ACCOUNT=production
```
"SLURM_ACCOUNT" is not used here, right?
```suggestion

```

Commit:
Answering Kalle's comments.

beatrizsavinhas commented:
### Copied flowcell to stage
`rsync -rvL /home/proj/production/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/ /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/`

### Testing demultiplexing with master
```terminal
rsync -rvL /home/proj/production/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/ /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/

```

Commit:
Troubleshooting unbound variable error.

karlnyr committed:
adding config

karlnyr committed:
add log

karlnyr committed:
variable def

karlnyr committed:
empty lines

beatrizsavinhas committed:
Update xdemuxtiles.bash

beatrizsavinhas commented:
Testing again

### Installing branch on stage
```Terminal
θ71° 13d [hiseq.clinical@hasta:/home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY] [S_base] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-demux-stage.sh fixing-demux-config-file
on the host hasta.scilifelab.se, continue
Collecting git+https://github.com/clinical-genomics/demultiplexing@fixing-demux-config-file
  Cloning https://github.com/clinical-genomics/demultiplexing (to revision fixing-demux-config-file) to /home/hiseq.clinical/tmp/pip-req-build-ghpzh1vk
  Running command git clone --quiet https://github.com/clinical-genomics/demultiplexing /home/hiseq.clinical/tmp/pip-req-build-ghpzh1vk
  Running command git checkout -b fixing-demux-config-file --track origin/fixing-demux-config-file
  Switched to a new branch 'fixing-demux-config-file'
  Branch fixing-demux-config-file set up to track remote branch fixing-demux-config-file from origin.
  Resolved https://github.com/clinical-genomics/demultiplexing to commit 6113a9f6ba292ecc7b971279eb1ec4238ae4970e
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: demux
  Building wheel for demux (pyproject.toml) ... done
  Created wheel for demux: filename=demux-5.11.0-py3-none-any.whl size=22745 sha256=710faab18927c8e68864603236939f42da54a33afe0e8eaae51b94c1053c1825
  Stored in directory: /home/hiseq.clinical/tmp/pip-ephem-wheel-cache-o3kw4y04/wheels/2b/47/33/3dc5f9c3c0116515664f5120b880a351d301445d0ce061bb8b
Successfully built demux
Installing collected packages: demux
  Attempting uninstall: demux
    Found existing installation: demux 5.11.0
    Uninstalling demux-5.11.0:
      Successfully uninstalled demux-5.11.0
Successfully installed demux-5.11.0
Collecting git+https://github.com/clinical-genomics/demultiplexing@fixing-demux-config-file
  Cloning https://github.com/clinical-genomics/demultiplexing (to revision fixing-demux-config-file) to /home/hiseq.clinical/tmp/pip-req-build-9c4nq56_
  Running command git clone --quiet https://github.com/clinical-genomics/demultiplexing /home/hiseq.clinical/tmp/pip-req-build-9c4nq56_
  Running command git checkout -b fixing-demux-config-file --track origin/fixing-demux-config-file
  Switched to a new branch 'fixing-demux-config-file'
  Branch fixing-demux-config-file set up to track remote branch fixing-demux-config-file from origin.
  Resolved https://github.com/clinical-genomics/demultiplexing to commit 6113a9f6ba292ecc7b971279eb1ec4238ae4970e
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: PyYAML<6.0.0,>=5.3.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (5.4.1)
Requirement already satisfied: bs4<0.0.2,>=0.0.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (0.0.1)
Requirement already satisfied: cglims<2.0.0,>=1.3.4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (1.3.4)
Requirement already satisfied: click<7.0.0,>=6.7.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (6.7)
Requirement already satisfied: importlib-metadata<4.0.0,>=3.1.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (3.10.1)
Requirement already satisfied: path.py<13.0.0,>=12.5.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (12.5.0)
Requirement already satisfied: psutil<6.0.0,>=5.7.3 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (5.9.4)
Requirement already satisfied: six<2.0.0,>=1.15.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (1.16.0)
Requirement already satisfied: snapshottest<0.7.0,>=0.6.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from demux==5.11.0) (0.6.0)
Requirement already satisfied: beautifulsoup4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from bs4<0.0.2,>=0.0.1->demux==5.11.0) (4.12.0)
Requirement already satisfied: genologics in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cglims<2.0.0,>=1.3.4->demux==5.11.0) (0.4.6)
Requirement already satisfied: python-dateutil in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cglims<2.0.0,>=1.3.4->demux==5.11.0) (2.8.2)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cglims<2.0.0,>=1.3.4->demux==5.11.0) (15.0.1)
Requirement already satisfied: zipp>=0.5 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from importlib-metadata<4.0.0,>=3.1.1->demux==5.11.0) (3.15.0)
Requirement already satisfied: path in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from path.py<13.0.0,>=12.5.0->demux==5.11.0) (16.6.0)
Requirement already satisfied: termcolor in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from snapshottest<0.7.0,>=0.6.0->demux==5.11.0) (2.2.0)
Requirement already satisfied: fastdiff<1,>=0.1.4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from snapshottest<0.7.0,>=0.6.0->demux==5.11.0) (0.3.0)
Requirement already satisfied: wasmer>=1.0.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from fastdiff<1,>=0.1.4->snapshottest<0.7.0,>=0.6.0->demux==5.11.0) (1.1.0)
Requirement already satisfied: wasmer-compiler-cranelift>=1.0.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from fastdiff<1,>=0.1.4->snapshottest<0.7.0,>=0.6.0->demux==5.11.0) (1.1.0)
Requirement already satisfied: soupsieve>1.2 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from beautifulsoup4->bs4<0.0.2,>=0.0.1->demux==5.11.0) (2.4)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from coloredlogs->cglims<2.0.0,>=1.3.4->demux==5.11.0) (10.0)
Requirement already satisfied: requests in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from genologics->cglims<2.0.0,>=1.3.4->demux==5.11.0) (2.28.2)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from requests->genologics->cglims<2.0.0,>=1.3.4->demux==5.11.0) (3.0.1)
Requirement already satisfied: idna<4,>=2.5 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from requests->genologics->cglims<2.0.0,>=1.3.4->demux==5.11.0) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from requests->genologics->cglims<2.0.0,>=1.3.4->demux==5.11.0) (1.26.14)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from requests->genologics->cglims<2.0.0,>=1.3.4->demux==5.11.0) (2022.12.7)
Collecting git+https://github.com/clinical-genomics/cgstats
  Cloning https://github.com/clinical-genomics/cgstats to /home/hiseq.clinical/tmp/pip-req-build-bs9uk7z2
  Running command git clone --quiet https://github.com/clinical-genomics/cgstats /home/hiseq.clinical/tmp/pip-req-build-bs9uk7z2
  Resolved https://github.com/clinical-genomics/cgstats to commit 168cccdb8d849767b140a5bffb3e19eb413e147b
  Preparing metadata (setup.py) ... done
Requirement already satisfied: Flask>=1.1.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (1.1.4)
Requirement already satisfied: Flask-Alchy in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.5.0)
Requirement already satisfied: Flask-Bootstrap in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (3.3.7.1)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (2.1)
Requirement already satisfied: PyYAML>=5.3.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (5.4.1)
Requirement already satisfied: SQLAlchemy in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (1.3.24)
Requirement already satisfied: alchy in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (2.2.2)
Requirement already satisfied: alembic in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (1.9.4)
Requirement already satisfied: bs4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.0.1)
Requirement already satisfied: click<7 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (6.7)
Requirement already satisfied: path.py in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (12.5.0)
Requirement already satisfied: pymysql in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (1.0.2)
Requirement already satisfied: bumpversion in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.6.0)
Requirement already satisfied: pytest in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (7.2.1)
Requirement already satisfied: wheel in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.38.4)
Requirement already satisfied: git-lint in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.1.2)
Requirement already satisfied: pylint in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (2.17.0)
Requirement already satisfied: pycodestyle in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (2.10.0)
Requirement already satisfied: yamllint in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (1.29.0)
Requirement already satisfied: docutils in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.19)
Requirement already satisfied: html-linter in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (0.4.0)
Requirement already satisfied: six>=1.12 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from cgstats==1.8.1) (1.16.0)
Requirement already satisfied: Werkzeug<2.0,>=0.15 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Flask>=1.1.1->cgstats==1.8.1) (0.16.1)
Requirement already satisfied: Jinja2<3.0,>=2.10.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Flask>=1.1.1->cgstats==1.8.1) (2.11.3)
Requirement already satisfied: itsdangerous<2.0,>=0.24 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Flask>=1.1.1->cgstats==1.8.1) (1.1.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from alchy->cgstats==1.8.1) (6.0.2)
Requirement already satisfied: Mako in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from alembic->cgstats==1.8.1) (1.2.4)
Requirement already satisfied: beautifulsoup4 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from bs4->cgstats==1.8.1) (4.12.0)
Requirement already satisfied: bump2version in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from bumpversion->cgstats==1.8.1) (1.0.1)
Requirement already satisfied: dominate in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Flask-Bootstrap->cgstats==1.8.1) (2.7.0)
Requirement already satisfied: visitor in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Flask-Bootstrap->cgstats==1.8.1) (0.1.3)
Requirement already satisfied: termcolor in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from git-lint->cgstats==1.8.1) (2.2.0)
Requirement already satisfied: docopt in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from git-lint->cgstats==1.8.1) (0.6.2)
Requirement already satisfied: template-remover in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from git-lint->cgstats==1.8.1) (0.1.9)
Requirement already satisfied: pathlib2 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from git-lint->cgstats==1.8.1) (2.3.7.post1)
Requirement already satisfied: path in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from path.py->cgstats==1.8.1) (16.6.0)
Requirement already satisfied: platformdirs>=2.2.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pylint->cgstats==1.8.1) (3.1.1)
Requirement already satisfied: astroid<=2.17.0-dev0,>=2.15.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pylint->cgstats==1.8.1) (2.15.0)
Requirement already satisfied: isort<6,>=4.2.5 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pylint->cgstats==1.8.1) (5.12.0)
Requirement already satisfied: mccabe<0.8,>=0.6 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pylint->cgstats==1.8.1) (0.7.0)
Requirement already satisfied: tomlkit>=0.10.1 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pylint->cgstats==1.8.1) (0.11.6)
Requirement already satisfied: dill>=0.3.6 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pylint->cgstats==1.8.1) (0.3.6)
Requirement already satisfied: attrs>=19.2.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pytest->cgstats==1.8.1) (22.2.0)
Requirement already satisfied: iniconfig in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pytest->cgstats==1.8.1) (2.0.0)
Requirement already satisfied: packaging in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pytest->cgstats==1.8.1) (23.0)
Requirement already satisfied: pluggy<2.0,>=0.12 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from pytest->cgstats==1.8.1) (1.0.0)
Requirement already satisfied: pathspec>=0.5.3 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from yamllint->cgstats==1.8.1) (0.11.1)
Requirement already satisfied: setuptools in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from yamllint->cgstats==1.8.1) (67.3.1)
Requirement already satisfied: lazy-object-proxy>=1.4.0 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from astroid<=2.17.0-dev0,>=2.15.0->pylint->cgstats==1.8.1) (1.9.0)
Requirement already satisfied: wrapt<2,>=1.14 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from astroid<=2.17.0-dev0,>=2.15.0->pylint->cgstats==1.8.1) (1.15.0)
Requirement already satisfied: MarkupSafe>=0.23 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from Jinja2<3.0,>=2.10.1->Flask>=1.1.1->cgstats==1.8.1) (2.0.1)
Requirement already satisfied: soupsieve>1.2 in /home/proj/stage/bin/miniconda3/envs/S_base/lib/python3.11/site-packages (from beautifulsoup4->bs4->cgstats==1.8.1) (2.4)
remote: Enumerating objects: 17, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 10 (delta 8), reused 5 (delta 4), pack-reused 0
Unpacking objects: 100% (10/10), done.
From https://github.com/Clinical-Genomics/demultiplexing
   245e885..6113a9f  fixing-demux-config-file -> origin/fixing-demux-config-file
Already on 'fixing-demux-config-file'
Your branch is behind 'origin/fixing-demux-config-file' by 2 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating 245e885..6113a9f
Fast-forward
 demux/utils/samplesheet.py      | 4 +++-
 scripts/hiseqx/xdemuxtiles.bash | 4 ++--
 2 files changed, 5 insertions(+), 3 deletions(-)
/home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY
No broken requirements found.
demux, version 5.11.0
cgstats, version 1.8.1
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY
done.
Log deploy... done.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY
done.
Log deploy... done.
```

###Testing demultiplexing with master
```Terminal
13d [hiseq.clinical@hasta:/home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY] [S_base] $ bash /home/proj/stage/bin/git/demultiplexing/scripts/hiseqx/xdemuxtiles.bash /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY
[20230329101732] demuxtiles.bash VERSION 5.5.0
[20230329101732] /home/proj/stage/bin/miniconda3/bin/conda run --name S_demux /home/proj/stage/bin/miniconda3/envs/S_demux/bin/demux -c /home/proj/stage/servers/config/hasta.scilifelab.se/demux-stage.yaml sheet fetch -a wgs  HLCKNCCXY > /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/SampleSheet.csv
[20230329101748] /home/proj/stage/bin/miniconda3/bin/conda run --name S_demux /home/proj/stage/bin/miniconda3/envs/S_demux/bin/demux -c /home/proj/stage/servers/config/hasta.scilifelab.se/demux-stage.yaml sheet validate --application wgs /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/SampleSheet.csv
ERROR conda.cli.main_run:execute(41): `conda run /home/proj/stage/bin/miniconda3/envs/S_demux/bin/demux -c /home/proj/stage/servers/config/hasta.scilifelab.se/demux-stage.yaml sheet validate --application wgs /home/proj/stage/flowcells/hiseqx/180522_ST-E00198_0301_BHLCKNCCXY/SampleSheet.csv` failed. (See above for error)
Traceback (most recent call last):
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/bin/demux", line 8, in <module>
    sys.exit(main())
  File "/home/proj/stage/bin/git/demultiplexing/demux/__main__.py", line 13, in main
    sys.exit(demux())
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/stage/bin/miniconda3/envs/S_demux/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/stage/bin/git/demultiplexing/demux/cli/samplesheet.py", line 49, in validate
    HiSeqXSamplesheet(samplesheet).validate()
  File "/home/proj/stage/bin/git/demultiplexing/demux/utils/samplesheet.py", line 341, in validate
    Samplesheet.validate(self)
  File "/home/proj/stage/bin/git/demultiplexing/demux/utils/samplesheet.py", line 309, in validate
    rs = _validate_unique_index(self.samplesheet)
  File "/home/proj/stage/bin/git/demultiplexing/demux/utils/samplesheet.py", line 276, in _validate_unique_index
    lanes = list(set(self.column("lane")))
  File "/home/proj/stage/bin/git/demultiplexing/demux/utils/samplesheet.py", line 192, in column
    yield line[column]
KeyError: 'lane'

ls: cannot access /home/proj/stage/demultiplexed-runs/180522_ST-E00198_0301_BHLCKNCCXY//projectlog.*.log: No such file or directory
```

karlnyr committed:
testing a new print

karlnyr committed:
Merge branch 'fixing-demux-config-file' of github.com:Clinical-Genomics/demultiplexing into fixing-demux-config-file

karlnyr committed:
lint

beatrizsavinhas committed:
Update samplesheet.py

karlnyr committed:
linting

karlnyr committed:
revert config calls and env management

karlnyr committed:
remova conda stuff, expand aliases

karlnyr committed:
version update

karlnyr committed:
source

karlnyr committed:
can we remove conda activation?

karlnyr committed:
env variables

karlnyr reviewed (approved):
fastet PR in the west

karlnyr committed:
env vars stage script
